src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,265 @@
+static_assert(PAL_GSS_C_EXTENDED_ERROR_FLAG == GSS_C_EXTENDED_ERROR_FLAG, "");
+static_assert(PAL_GSS_C_DELEG_POLICY_FLAG == GSS_C_DELEG_POLICY_FLAG, "");
+
+static_assert(PAL_GSS_COMPLETE == GSS_S_COMPLETE, "");
+static_assert(PAL_GSS_CONTINUE_NEEDED == GSS_S_CONTINUE_NEEDED, "");
+
+#if !HAVE_GSS_SPNEGO_MECHANISM
+static char gss_mech_value[] = "\x2b\x06\x01\x05\x05\x02"; // Binary representation of SPNEGO Oid (RFC 4178)
+#endif
+
+static void NetSecurityNative_HandleError(uint32_t majorStatus, GssBuffer** outBufferHandle, uint32_t* outBufferLength)
+{
+    assert(outBufferHandle != nullptr);
+    assert(outBufferLength != nullptr);
+    if (GSS_ERROR(majorStatus) || (*outBufferHandle == nullptr) ||
+        (*outBufferHandle)->length > UnsignedCast(std::numeric_limits<uint32_t>::max()))
+    {
+        uint32_t relBufferStatus;
+        gss_release_buffer(&relBufferStatus, *outBufferHandle);
+        *outBufferHandle = nullptr;
+        *outBufferLength = 0;
+    }
+    else
+    {
+        *outBufferLength = static_cast<uint32_t>((*outBufferHandle)->length);
+    }
+}
+
+extern "C" uint32_t NetSecurityNative_AcquireCredSpNego(uint32_t* minorStatus,
+                                                        GssName* desiredName,
+                                                        int32_t isInitiate,
+                                                        GssCredId** outputCredHandle)
+{
+    assert(minorStatus != nullptr);
+    assert(outputCredHandle != nullptr);
+    assert(isInitiate == 0 || isInitiate == 1);
+#if HAVE_GSS_SPNEGO_MECHANISM
+    gss_OID_set_desc gss_mech_spnego_OID_set_desc = {.count = 1, .elements = GSS_SPNEGO_MECHANISM};
+#else
+    gss_OID_desc gss_mech_spnego_OID_desc = {.length = 6, .elements = static_cast<void*>(gss_mech_value)};
+    gss_OID_set_desc gss_mech_spnego_OID_set_desc = {.count = 1, .elements = &gss_mech_spnego_OID_desc};
+#endif
+    gss_cred_usage_t credUsage = isInitiate ? GSS_C_INITIATE : GSS_C_ACCEPT;
+    return gss_acquire_cred(
+        minorStatus, desiredName, 0, &gss_mech_spnego_OID_set_desc, credUsage, outputCredHandle, nullptr, nullptr);
+}
+
+extern "C" uint32_t NetSecurityNative_DeleteSecContext(uint32_t* minorStatus, GssCtxId** contextHandle)
+{
+    assert(contextHandle != nullptr);
+    assert(minorStatus != nullptr);
+    return gss_delete_sec_context(minorStatus, contextHandle, GSS_C_NO_BUFFER);
+}
+
+extern "C" uint32_t NetSecurityNative_DisplayStatus(uint32_t* minorStatus,
+                                                    uint32_t statusValue,
+                                                    int32_t isGssMechCode,
+                                                    GssBuffer** outBufferHandle,
+                                                    uint32_t* statusLength)
+{
+    assert(minorStatus != nullptr);
+    assert(outBufferHandle != nullptr);
+    assert(statusLength != nullptr);
+    assert(isGssMechCode == 0 || isGssMechCode == 1);
```

Comment:
statusValue?  It'd also be nice if the asserts were in the same order as the arguments.